### PR TITLE
fix(cxx_extractor): don't crash when modules are on

### PR DESCRIPTION
**Pull request by zrlk**
In service of #5543.

This PR should only affect the behavior of the extractor where it would have either crashed because of a null dereference or due to a CHECK-fail. The conditions leading to these cases occur when modules are turned on.

### CODE REVIEW

**Review by shahms — COMMENTED** [review]
Please add a test for this.

**Comment by zrlk** [comment]
It appears that the OSS Bazel rules don't support header-only modules. An internal test would probably work; otherwise I think we're looking at a sh test with the cxx_extractor binary (ugh) or a hand-edited xa (ugh).  I would be happy to be proven incorrect.

**Comment by shahms** [comment]
> 

None of the existing extractor integration tests use the Bazel rules or shell, see, for example: https://github.com/kythe/kythe/blob/master/kythe/cxx/extractor/testdata/root_directory_test.cc

Is there some reason that approach won't work here?

**Comment by zrlk** [comment]
OK, the new test hits the new code path. I expect to refine the set of flags as I dig deeper into this. (also ugh `-Xclang`)

File: kythe/cxx/extractor/cxx_extractor.cc [hunk 1]
```diff
@@ -591,8 +619,10 @@ PreprocessorTranscript ExtractorPPCallbacks::PopFile() {
 }
 
 void ExtractorPPCallbacks::EndOfMainFile() {
-  AddFile(GetMainFile(), std::string(GetMainFile()->getName()));
-  *main_source_file_transcript_ = PopFile();
+  if (auto* mfile = GetMainFile()) {
+    AddFile(GetMainFile(), std::string(GetMainFile()->getName()));
```
Comment (shahms) [inline kythe/cxx/extractor/cxx_extractor.cc:623]: `s/GetMainFile()/mfile/g` on this line?

**Review by shahms — APPROVED** [review]
Thanks!

**Pull request merged**